src/_pytest/python.py

```diff
@@ -132,7 +132,25 @@ def pytest_generate_tests(metafunc):
     _validate_parametrize_spelling(metafunc)
 
     for marker in metafunc.definition.iter_markers(name="parametrize"):
-        metafunc.parametrize(*marker.args, **marker.kwargs)
+        args, kwargs = marker.args, marker.kwargs
+        if marker._param_ids_from:
+            ids = marker._param_ids_from._param_ids_generated
+            if ids is None:
+                save_ids = True
+            else:
+                save_ids = False
+                if len(args) > 3:
+                    args = args.copy()
+                    args[3] = ids
+                else:
+                    assert "ids" in kwargs
+                    kwargs = kwargs.copy()
+                    kwargs["ids"] = ids
```

Comment:
Having ids either as arg or kwarg is also not nice here.
Could be combined with `_has_param_ids` though at least.